.github/autolabeler.yml

```diff
@@ -0,0 +1,54 @@
+#
+# Licensed to the Apache Software Foundation (ASF) under one or more
+# contributor license agreements.  See the NOTICE file distributed with
+# this work for additional information regarding copyright ownership.
+# The ASF licenses this file to You under the Apache License, Version 2.0
+# (the "License"); you may not use this file except in compliance with
+# the License.  You may obtain a copy of the License at
+#
+#    https://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+# Bot page: https://github.com/apps/probot-autolabeler
+# The matching patterns follow the .gitignore spec.
+# See: https://git-scm.com/docs/gitignore#_pattern_format
+
+infra:
+  - ".github/"
+  - "appveyor.yml"
+  - "/tools/"
+build:
+  - "/dev/"
+  - "/build/"
+  - "/project/"
+release:
+  - "/dev/create-release/"
+docs:
+  - "docs/"
+  - "examples/"
+  - "/README.md"
+  - "/CONTRIBUTING.md"
+core:
+  - "/core/"
+sql:
+  - "sql/"
+ml:
+  - "ml/"
+  - "mllib/"
+  - "mllib-local/"
+streaming:
+  - "streaming/"
+python:
+  - "python/"
+java:
+  - "/common/"
+  - "java/"
+R:
+  - "r/"
```

Comment:
It think we should also add `/r/` because some files like `sql/core/src/main/scala/org/apache/spark/sql/api/r/`.

Reply:
`/r/` will only match `r/` at the root of the project. `r/` will match a directory named `r` anywhere in the folder hierarchy.